### PR TITLE
Fix Prometheus PVC Permissions

### PR DESCRIPTION
**Pull request by dvaldivia**
Fixes the permissions on the PVC to address the problem discussed at 

https://github.com/prometheus-operator/prometheus-operator/issues/3459
https://github.com/prometheus-operator/prometheus-operator/issues/966

The init container approached worked, which I think it's better to running the container in a [privileged mode](https://github.com/prometheus-operator/prometheus-operator/issues/966#issuecomment-365037713)

### CODE REVIEW

File: pkg/resources/statefulsets/prometheus.go [hunk 1]
```diff
@@ -59,7 +59,8 @@ func prometheusEnvVars(t *miniov1.Tenant) []corev1.EnvVar {
 func prometheusConfigVolumeMount(t *miniov1.Tenant) corev1.VolumeMount {
 	return corev1.VolumeMount{
 		Name:      t.PrometheusConfigVolMountName(),
-		MountPath: "/etc/prometheus",
+		MountPath: "/etc/prometheus/prometheus.yml",
+		SubPath:   "prometheus.yml",
```
Comment (donatello) [inline pkg/resources/statefulsets/prometheus.go:63]:
```suggestion
		SubPath:   "prometheus",
```

Right?

Comment (dvaldivia) [reply pkg/resources/statefulsets/prometheus.go:63]: This makes it so I only mount the `prometheus.yml` from the configmap into `/etc/prometheus/prometheus.yml`, if we switch to subpath `prometheus` the yml file will be mounted as a extensionless file

Comment (donatello) [reply pkg/resources/statefulsets/prometheus.go:63]: Ok sounds good.

**Review by donatello — DISMISSED** [review]

File: pkg/resources/statefulsets/prometheus.go [hunk 2]
```diff
@@ -129,6 +146,31 @@ func NewForPrometheus(t *miniov1.Tenant, serviceName string) *appsv1.StatefulSet
...
+	//  - name: "init-chown-data"
+	//    image: "busybox"
+	//    # 1000 is the user that prometheus uses.
+	//    command: ["chown", "-R", "1000:2000", /var/prometheus/data]
+	//    volumeMounts:
+	//    - name: prometheus-kube-prometheus-db
+	//      mountPath: /var/prometheus/data
+	initContainers := []corev1.Container{
+		{
+			Name:  "init-chown-data",
```
Comment (krisis) [inline pkg/resources/statefulsets/prometheus.go:162]: nit: `init-chown-data` may alarm users browsing k8s workloads on tenant's namespace. Prefixing container name with `prometheus` is aimed at clarifying init container's  purpose/scope.
```suggestion
			Name:  "prometheus-init-chown-data",
```

Comment (dvaldivia) [reply pkg/resources/statefulsets/prometheus.go]: done

**Review by krisis — APPROVED** [review]

**Review by donatello — APPROVED** [review]

**Pull request merged**